types/pipeline_events.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "time"
+
+type PipelineEventSystem string
+
+const (
+	PipelineEventSystemDeployment PipelineEventSystem = "k8s:deployment"
+	PipelineEventSystemPod        PipelineEventSystem = "k8s:pod"
+)
+
+// PipelineEvent model.
+type PipelineEvent struct {
+	PipelineID string              `json:"pipelineID" yaml:"pipelineID"`
+	System     PipelineEventSystem `json:"system" yaml:"system"`
+	Status     PipelineStatusKind  `json:"kind" yaml:"kind"`
+	Reason     string              `json:"reason" yaml:"reason"`
+	Message    string              `json:"message" yaml:"message"`
+	CreatedAt  time.Time           `json:"createdAt" yaml:"createdAt"`
+}
+
+// CreatePipelineEvents is the format we submit the events in.
+type CreatePipelineEvents []PipelineEvent
+
+// CreatePipelineEventsResponse is the response we get when creating new pipeline
+// events.
+type CreatePipelineEventsResponse struct {
```

Comment:
```suggestion
type CreatedPipelineEvents struct {
```

Reply:
But really, I don't understand the response here. I mean, its inserting an array of events...
Maybe it could simply return a `204 No Content` 🤔

Reply:
Will the client API as it is now allow that?

Reply:
No, it doesn't need a response body. We have plenty of update and delete endpoints that just return `204 No Content` without body at all.
In the create endpoints, we normally return with the generated id, and createdAt. But really any useful field could be returned.

Reply:
I return the StatusID now while creating a single event.